### PR TITLE
Update version, changelog, and release notes for 0.9.0-rc1 release 

### PR DESCRIPTION
**Pull request by alerickson**
# PR Summary
Update version, changelog, and release notes for 0.9.0-rc1 release 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary.

### CODE REVIEW

File: src/Microsoft.PowerShell.PSResourceGet.psd1 [hunk 1]
```diff
@@ -45,7 +45,7 @@
         'udres')
     PrivateData = @{
         PSData = @{
-            Prerelease   = 'beta24'
+            Prerelease   = 'rc'
```
Comment (ThomasNieto) [inline src/Microsoft.PowerShell.PSResourceGet.psd1]:
```suggestion
            Prerelease   = 'rc1'
```
Would it be better to name it `rc1` if there is a second `rc`?

**Review by ThomasNieto — CHANGES_REQUESTED** [review]

File: .ci/ci_release.yml [hunk 2]
```diff
@@ -292,7 +292,7 @@ stages:
           BuildDropPath: $(signOutPath)
           Build_Repository_Uri: 'https://github.com/powershell/PSResourceGet'
           PackageName: 'Microsoft.PowerShell.PSResourceGet'
-          PackageVersion: '0.5.24-beta24'
+          PackageVersion: '0.9.0-rc'
```
Comment (alerickson) [inline .ci/ci_release.yml]:
```suggestion
          PackageVersion: '0.9.0-rc1'
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,16 @@
 # CHANGELOG
+## 0.9.0-rc
```
Comment (alerickson) [inline CHANGELOG.md]:
```suggestion
## 0.9.0-rc1
```

File: src/Microsoft.PowerShell.PSResourceGet.psd1 [hunk 4]
```diff
@@ -55,6 +55,18 @@
             ProjectUri   = 'https://go.microsoft.com/fwlink/?LinkId=828955'
             LicenseUri   = 'https://go.microsoft.com/fwlink/?LinkId=829061'
             ReleaseNotes = @'
+## 0.9.0-rc
```
Comment (alerickson) [inline src/Microsoft.PowerShell.PSResourceGet.psd1]:
```suggestion
## 0.9.0-rc1
```

**Review by SydneyhSmith — APPROVED** [review]

**Review by anamnavi — APPROVED** [review]
LGTM

**Pull request merged**